salt/pillar/mysql.py

```diff
@@ -59,16 +59,29 @@
 
 # Import third party libs
 try:
+    # Trying to import MySQLdb
     import MySQLdb
-    HAS_MYSQL = True
+    import MySQLdb.cursors
+    import MySQLdb.converters
+    import MySQLdb.constants
 except ImportError:
-    HAS_MYSQL = False
+    try:
+        # MySQLdb import failed, try to import PyMySQL
+        import pymysql
+        pymysql.install_as_MySQLdb()
+        import MySQLdb
+        import MySQLdb.cursors
+        import MySQLdb.converters
+        import MySQLdb.constants
+    except ImportError:
+        MySQLdb = None
 
 
 def __virtual__():
-    if not HAS_MYSQL:
-        return False
-    return True
+    '''
+    Confirm that a python mysql client is installed.
+    '''
+    return (bool(MySQLdb), "No python mysql client installed." if MySQLdb is None else '')
```

Comment:
And these too. Also for consistency sake, use double-quotes only when needed, say you want show inside the message single quotes etc. But normally we use single quotes all the time. :wink: